pyomo/contrib/pynumero/interfaces/nlp_projections.py

```diff
@@ -0,0 +1,303 @@
+from pyomo.contrib.pynumero.interfaces.nlp import NLP
+import numpy as np
+import scipy.sparse as sp
+
+class _BaseNLPDelegator(NLP):
```

Comment:
We should add an issue to revisit this class and have an implementation that is a little less fragile.

Reply:
done.